packages/nextra-theme-docs/src/components/search.tsx

```diff
@@ -85,121 +79,35 @@ export function Search({
-          if (active - 1 >= 0) {
-            const el = ulRef.current?.querySelector<HTMLAnchorElement>(
-              `li:nth-of-type(${active}) > a`
-            )
-            if (el) {
-              e.preventDefault()
-              handleActive({ currentTarget: el })
-              el.focus()
-            }
-          }
-          break
-        }
-        case 'Enter': {
-          const result = results[active]
-          if (result) {
-            void router.push(result.route)
-            finishSearch()
-          }
-          break
-        }
-        case 'Escape': {
-          setShow(false)
-          input.current?.blur()
-          break
-        }
-      }
-    },
-    [active, results, router, finishSearch, handleActive]
-  )
-
-  const mounted = useMounted()
-  const renderList = show && Boolean(value)
-
-  const icon = (
-    <Transition
-      show={mounted && (!show || Boolean(value))}
-      enter="_transition-opacity"
-      enterFrom="_opacity-0"
-      enterTo="_opacity-100"
-      leave="_transition-opacity"
-      leaveFrom="_opacity-100"
-      leaveTo="_opacity-0"
+  const icon = mounted && !focused && (
+    <kbd
+      className={cn(
+        '_absolute _my-1.5 _select-none ltr:_right-1.5 rtl:_left-1.5',
+        '_h-5 _rounded _bg-white _px-1.5 _font-mono _text-[11px] _font-medium _text-gray-500',
+        '_border dark:_border-gray-100/20 dark:_bg-black/50',
+        'contrast-more:_border-current contrast-more:_text-current contrast-more:dark:_border-current',
+        '_items-center _gap-1 _flex',
+        'max-sm:_hidden'
+      )}
     >
-      <kbd
-        className={cn(
-          '_absolute _my-1.5 _select-none ltr:_right-1.5 rtl:_left-1.5',
-          '_h-5 _rounded _bg-white _px-1.5 _font-mono _text-[10px] _font-medium _text-gray-500',
-          '_border dark:_border-gray-100/20 dark:_bg-black/50',
-          'contrast-more:_border-current contrast-more:_text-current contrast-more:dark:_border-current',
-          '_items-center _gap-1 _transition-opacity _flex',
-          value
-            ? '_z-20 _cursor-pointer hover:_opacity-70'
-            : '_pointer-events-none max-sm:_hidden'
-        )}
-        title={value ? 'Clear' : undefined}
-        onClick={clearValue}
-      >
-        {value && focused
-          ? 'ESC'
-          : mounted &&
-            (navigator.userAgent.includes('Macintosh') ? (
-              <>
-                <span className="_text-xs">⌘</span>K
-              </>
-            ) : (
-              'CTRL K'
-            ))}
-      </kbd>
-    </Transition>
+      {navigator.userAgent.includes('Macintosh') ? (
+        <>
+          <span className="_text-xs">⌘</span>K
+        </>
+      ) : (
+        'CTRL K'
+      )}
+    </kbd>
   )
 
+  const handleKeyDown = useCallback(<T,>(event: KeyboardEvent<T>) => {
+    // skip the character selection process in the IME for CJK language users.
+    if (event.nativeEvent.isComposing) return
+  }, [])
```

Comment:
@87xie I added this, lmk maybe it fixes your issue

also I pushed another commit https://github.com/shuding/nextra/pull/3186/commits/3f93edfb14475540cd12381c25321b7cf44bdc97 where I fix cmdk+k shortcut (accidentally removed) and I removed `ESC` icon in input because it's doesn't work since headlessui adds `inert` attribute to esc icon once we type something

Reply:
It cannot prevent headlessui's `onKeyDown` callback from continuing to execute, so we can remove it for now.